internal/k8s/crdcontroller.go

```diff
@@ -208,3 +219,214 @@ func (c *AviController) SetupAKOCRDEventHandlers(numWorkers uint32) {
 
 	return
 }
+
+// validateHostRuleObj would do validation checks
+// update internal CRD caches, and push relevant ingresses to ingestion
+func ValidateHostRuleObj(key string, hostrule *akov1alpha1.HostRule) error {
+	var err error
+	fqdn := hostrule.Spec.VirtualHost.Fqdn
+	foundHost, foundHR := objects.SharedCRDLister().GetFQDNToHostruleMapping(fqdn)
+	if foundHost && foundHR != hostrule.Namespace+"/"+hostrule.Name {
+		err = fmt.Errorf("duplicate fqdn %s found in %s", fqdn, foundHR)
+		status.UpdateHostRuleStatus(key, hostrule, status.UpdateCRDStatusOptions{
+			Status: lib.StatusRejected,
+			Error:  err.Error(),
+		})
+		utils.AviLog.Warnf("key: %s, msg: %v", key, err)
```

Comment:
Is there a reason for this warning here?

Reply:
Removed.